### DISCUSSION TITLE
Uncaught TypeError: Cannot read properties of null (reading 'title') at Gp export

### DISCUSSION THREAD

**Discussion by hegelespaul** [CoderLine/alphaTab#993] [Q&A] [answered]
\title "My Song" 
\subtitle "Me"
\tempo 30
.
(4.3 1.5 4.4 4.2 ).1{ch "SomeChord"} |(2.3 4.4 3.2 3.5 ).1{ch "SomeChord"} |(4.3 5.4 2.6 3.2 ).1{ch "SomeChord"} |(3.3 4.6 2.4 3.2 ).1{ch "SomeChord"} |(3.3 3.6 1.4 3.2 ).1{ch "SomeChord"} |(2.3 3.6 2.4 1.2 ).1{ch "SomeChord"} |(2.3 2.6 5.5 1.2 ).1{ch "SomeChord"} |(4.3 3.6 4.4 3.2 ).1{ch "SomeChord"} |(3.3 4.6 2.4 3.2 ).1{ch "SomeChord"} |(3.3 3.6 1.4 3.2 ).1{ch "SomeChord"} |(3.3 2.6 2.4 2.2 ).1{ch "SomeChord"} |(4.3 6.6 4.4 4.2 ).1{ch "SomeChord"} 

I have this text file in a blob URL that I try to export with this code (a copy of the one on the page):

const exporter = new alphaTab.exporter.Gp7Exporter();
const data = exporter.export(api.score, api.settings); // will return a Uint8Array

// trigger download
const a = document.createElement('a');
a.download = api.score.title.length > 0 ? api.score.title + '.gp' : 'Untitled.gp';
a.href = URL.createObjectURL(new Blob([data]));
document.body.appendChild(a);
a.click();
document.body.removeChild(a);

Does anybody have an idea why it returns me this error:

alphaTab.js:40180 
        
       Uncaught TypeError: Cannot read properties of null (reading 'title')
    at GpifWriter.writeScoreNode (alphaTab.js:40180:58)
    at GpifWriter.writeDom (alphaTab.js:39646:18)
    at GpifWriter.writeXml (alphaTab.js:39633:18)
    at Gp7Exporter.writeScore (alphaTab.js:42451:40)
    at Gp7Exporter.export (alphaTab.js:39574:18)
    at compgen.js:613:29

I'm newbie with alphatab and really could use some help.

Thanks!

**Comment by Danielku15** [comment, marked as answer]
Hard to tell with a full example what is happening here. I put it into a Fiddle and export seems to be working fine with https://jsfiddle.net/danielku15/4dgomebc/

From the error it seems that `api.score` might be actually null in your case when you are calling the exporter.

**Reply by hegelespaul** [reply]
Thanks, the code in the fiddle totally works, I just put my code in a function triggered by a button and it exports. This solution definitively works for me, Thanks!
